### ISSUE TITLE
[iOS/XCode] The Dart VM service protocol port for debugging does not match the attempted connection port from the `flutter run` CLI.

### ISSUE THREAD

**Issue by kNoAPP** [flutter/flutter#146709]
### Steps to reproduce

Primarily,
1. On MacOS XCode 15.3, create a sample Flutter app.
2. Run `flutter run` with a physical iOS device connected/attached.
3. Observe the issue: debugging is not possible.

Alternatively, use the following VSC `launch.json` configuration:
```json
        {
            "name": "Launch Flutter",
            "type": "dart",
            "request": "launch",
            "flutterMode": "debug",
            "cwd": "${workspaceFolder}",
        },
```

### Expected results

The Dart VM service protocol for debugging should listen to the same port as the connecting pair. 
For example: `http://127.0.0.1:49241/-hJkog_kS0Y=/`

### Actual results

The Dart VM service protocol port for debugging and the attempted connection port are different, resulting in an error and an inability to debug iOS devices.
For example:
From XCode:
`flutter: The Dart VM service is listening on http://127.0.0.1:49241/-hJkog_kS0Y=/`
and from the CLI:
`Error connecting to the service protocol: failed to connect to http://127.0.0.1:49426/-hJkog_kS0Y=/`

Port 49241 and port 49426 do not match.

### Code sample

<details open><summary>Code sample</summary>

Any Flutter application will do.

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![Screenshot 2024-04-12 165317](https://github.com/flutter/flutter/assets/23529044/48d21e89-5b99-4da6-93c2-015748a346a9)
![Screenshot 2024-04-12 165252](https://github.com/flutter/flutter/assets/23529044/39112617-7bfe-4046-846b-c3d75fcdd050)

</details>

### Logs

<details open><summary>Logs</summary>

```console
flutter run
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
alden@aldens-mac-mini hylist % flutter doctor -v
[✓] Flutter (Channel stable, 3.19.5, on macOS 14.4.1 23E224 darwin-arm64, locale en-US)
    • Flutter version 3.19.5 on channel stable at /Users/alden/development/flutter
    • Upstream repository https://github.com/flutter/flutter
    • Framework revision 300451adae (2 weeks ago), 2024-03-27 21:54:07 -0500
    • Engine revision e76c956498
    • Dart version 3.3.3
    • DevTools version 2.31.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/alden/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)

[✓] VS Code (version 1.88.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.86.0

[✓] Connected device (3 available)
    • Alden’s iPad (mobile) • 00008103-000949C90A88A01E • ios            • iOS 17.4.1 21E236
    • macOS (desktop)       • macos                     • darwin-arm64   • macOS 14.4.1 23E224 darwin-arm64
    • Chrome (web)          • chrome                    • web-javascript • Google Chrome 123.0.6312.122

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by huycozy** [comment]
Hi @kNoAPP 
It doesn't seem to be clear about the reproduction steps. I'm not sure which step should be done on XCode and which step on VSCode. Could you please share more details? If there is a recorded video, it would be helpful.

**Comment by kNoAPP** [comment]
Fixed in `v3.19.6`.
